test/unit/DCAStrategies/dca-strategies-permissions-manager.spec.ts

```diff
@@ -0,0 +1,201 @@
+        expect(await DCAPermissionsManager.mintCounter()).to.equal(0);
+      });
+    });
+  });
+
+  describe('block number', () => {
+    const BLOCK_NUMBER = 555888222;
+
+    given(async () => {
+      await DCAPermissionsManager.setBlockNumber(BLOCK_NUMBER);
+    });
+
+    when('getting block number', () => {
+      then('block number is correct', async () => {
+        expect(await DCAPermissionsManager.getBlockNumber()).to.equal(BLOCK_NUMBER);
+      });
+    });
+  });
+
+  describe('mint', () => {
+    let tokenId: number;
+    given(async () => {
+      tokenId = (await DCAPermissionsManager.mintCounter()).toNumber() + 1;
+    });
+    when('owner is zero address', () => {
+      then('reverts with message', async () => {
+        await behaviours.txShouldRevertWithMessage({
+          contract: DCAPermissionsManager,
+          func: 'mint',
+          args: [constants.ZERO_ADDRESS, []],
+          message: 'ERC721: mint to the zero address',
+        });
+      });
+    });
+    when('mint is executed', () => {
+      const OWNER = wallet.generateRandomAddress();
+      const OPERATOR = constants.NOT_ZERO_ADDRESS;
+      let tx: TransactionResponse;
+      let initialMintCounter: BigNumber;
+
+      given(async () => {
+        initialMintCounter = await DCAPermissionsManager.mintCounter();
+        tx = await DCAPermissionsManager.mint(OWNER, [{ operator: OPERATOR, permissions: [Permission.WITHDRAW] }]);
+      });
+
+      then('mint counter gets increased', async () => {
+        expect(await DCAPermissionsManager.mintCounter()).to.equal(initialMintCounter.add(1));
+      });
+
+      //   then('owner has all permisisons', async () => {
+      //     expect(await DCAPermissionsManager.hasPermission(TOKEN_ID, OWNER, Permission.INCREASE)).to.be.true;
+      //     expect(await DCAPermissionsManager.hasPermission(TOKEN_ID, OWNER, Permission.REDUCE)).to.be.true;
+      //     expect(await DCAPermissionsManager.hasPermission(TOKEN_ID, OWNER, Permission.TERMINATE)).to.be.true;
+      //     expect(await DCAPermissionsManager.hasPermission(TOKEN_ID, OWNER, Permission.WITHDRAW)).to.be.true;
+      //   });
+
+      //   then('permissions are assigned properly', async () => {
+      //     expect(await DCAPermissionsManager.hasPermission(TOKEN_ID, OPERATOR, Permission.WITHDRAW)).to.be.true;
+      //   });
+
+      //   then('no extra permissions are assigned', async () => {
+      //     for (const permission of [Permission.INCREASE, Permission.REDUCE, Permission.TERMINATE]) {
+      //       expect(await DCAPermissionsManager.hasPermission(TOKEN_ID, OPERATOR, permission)).to.be.false;
+      //     }
+      //   });
+
+      then('nft is created and assigned to owner', async () => {
+        const tokenOwner = await DCAPermissionsManager.ownerOf(tokenId);
+        const balance = await DCAPermissionsManager.balanceOf(OWNER);
+        expect(tokenOwner).to.equal(OWNER);
+        expect(balance).to.equal(tokenId);
```

Comment:
Balance returns the amount of owned NFTs by the owner, not the ids of the owned NFTs
```suggestion
        expect(balance).to.equal(1);
```